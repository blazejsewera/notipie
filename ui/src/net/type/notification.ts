export type Notification = {
  appName: string
  timestamp: string
  appImgUri?: string
  title: string
  subtitle?: string
  body?: string
  id?: string
  read?: boolean
  extUri?: string
  readUri?: string
  archiveUri?: string
}
