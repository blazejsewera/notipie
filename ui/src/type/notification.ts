export type Notification = {
  appName: string
  timestamp: string
  appImgUri?: string
  title: string
  subtitle?: string
  body?: string
}
