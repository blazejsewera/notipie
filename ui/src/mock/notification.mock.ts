import type { Notification } from '../type/notification'
import githubIcon from './asset/github-icon.svg'

const body = `#12 add some new amazing functionality

Closes #10. Changes both in 'core' and 'ui'. Needs additional work with this and that.`

export const full: Notification = {
  appName: 'Github',
  title: 'New Pull Request',
  subtitle: 'â€” notipie',
  body,
  timestamp: '2 hours ago',
  uuid: '1',
}

export const fullWithImage: Notification = {
  ...full,
  appImgUri: githubIcon,
}

export const partial: Notification = {
  appName: full.appName,
  title: full.title,
  body: full.body,
  timestamp: full.timestamp,
  uuid: '2',
}

export const minimal: Notification = {
  appName: full.appName,
  title: full.title,
  timestamp: full.timestamp,
  uuid: '3',
}
