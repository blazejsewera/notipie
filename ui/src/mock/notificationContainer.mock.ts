import { NotificationContainerHandlers } from '../type/handler'

export const handlers: NotificationContainerHandlers = {
  onCheckAll: () => {},
}
