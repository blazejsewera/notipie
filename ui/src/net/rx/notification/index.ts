import { catchError, map, merge, Observable, observableOf } from '../../../external/rxjs'
import { NotificationWithStatus } from '../../../store'
import { Notification } from '../../../type/notification'
import { addHandlers } from '../../../util/notification/postprocessor'
import { updateTime } from '../../../util/notification/time'
import { getExisting$ } from './getExisting'
import { receivePushed$ } from './receivePushed'

const toNotificationWithStatus = (notification: Notification): NotificationWithStatus => ({
  notification,
  status: 'ok',
})

export const notification$: Observable<NotificationWithStatus> = merge(getExisting$, receivePushed$).pipe(
  map(addHandlers),
  map(updateTime),
  map(toNotificationWithStatus),
  catchError(() => observableOf<NotificationWithStatus>({ status: 'error' })),
)
