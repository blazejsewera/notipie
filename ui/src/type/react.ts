import type { ReactElement, ReactNode } from 'react'

type EmptyObject = Record<string, never>
type Children = ReactNode

/**
 * No props, a.k.a. () => </>, or ({ children }) => </> component
 */
export type EmptyProps = EmptyObject

/**
 * No React Context
 */
export type EmptyContext = EmptyObject

/**
 * Functional Component without children
 */
export interface FC<P = EmptyProps, Ctx = EmptyContext> {
  (props: P, context?: Ctx): ReactElement | null
  displayName?: string
}

/**
 * Functional Component with children
 */
export interface FCC<C = Children, Ctx = EmptyContext> {
  (props: { children?: C }, context?: Ctx): ReactElement | null
  displayName?: string
}

/**
 * Functional Component with props and children
 */
export interface FCPC<P = EmptyProps, C = Children, Ctx = EmptyContext> {
  (props: P & { children?: C }, context?: Ctx): ReactElement | null
  displayName?: string
}
