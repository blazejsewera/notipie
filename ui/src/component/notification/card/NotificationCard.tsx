import * as React from 'react'
import type { FC } from '../../../type/react'
import type { Notification } from '../../../type/notification'
import { cx } from '../../../util/cx'
import { Footer } from './section/Footer'
import { Header } from './section/Header'
import { Indicator } from './sprite/icon/Indicator'
import { Body } from './text/Body'

export interface NotificationCardProps {
  notification: Notification
}

export const NotificationCard: FC<NotificationCardProps> = ({ notification }) => {
  const { appName, appImgUri, title, subtitle, body, timestamp } = notification

  return (
    <div className={cx('rounded-3xl', 'bg-white', 'w-80', 'p-5', 'shadow-md', 'relative')}>
      <Header {...{ appName, appImgUri, title, subtitle }} bgColor="darkgreen" />
      <Indicator />
      {body ? <Body>{body}</Body> : null}
      <Footer {...{ appName, timestamp }} />
    </div>
  )
}
