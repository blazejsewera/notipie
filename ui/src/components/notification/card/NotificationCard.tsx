import * as React from 'react'
import { cx } from '../../../../src/utils/cx'
import { Footer } from './section/Footer'
import { Header } from './section/Header'
import { Indicator } from './sprite/icon/Indicator'
import { Body } from './text/Body'

export interface NotificationCardProps {
  appName: string
  timestamp: string
  appImgUri?: string
  title: string
  subtitle?: string
  body?: string
}

export const NotificationCard: React.FC<NotificationCardProps> = ({
  appName,
  timestamp,
  appImgUri,
  title,
  subtitle,
  body,
}) => (
  <div className={cx('rounded-3xl', 'bg-white', 'w-80', 'p-5', 'shadow-md', 'relative')}>
    <Header {...{ appName, appImgUri, title, subtitle }} bgColor="darkgreen" />
    <Indicator />
    <Body>{body}</Body>
    <Footer {...{ appName, timestamp }} />
  </div>
)
