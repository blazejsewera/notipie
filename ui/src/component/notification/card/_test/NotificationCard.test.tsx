import * as React from 'react'
import type { Notification } from '../../../../type/notification'
import { NotificationCard } from '../NotificationCard'
import renderer from 'react-test-renderer'
import { intlMock } from '../../../../mock/intl.mock'
import { full, fullWithImage, minimal, partial } from '../../../../mock/notification.mock'

describe('NotificationCard component tests', () => {
  // given
  const intl = intlMock

  const testNotificationCard = (notification: Notification) => {
    const tree = renderer.create(<NotificationCard notification={notification} intl={intl} />).toJSON()
    expect(tree).toMatchSnapshot()
  }

  it('renders correctly with minimal notification', () => {
    testNotificationCard(minimal)
  })
  it('renders correctly with partial notification', () => {
    testNotificationCard(partial)
  })
  it('renders correctly with full notification', () => {
    testNotificationCard(full)
  })
  it('renders correctly with full with image notification', () => {
    testNotificationCard(fullWithImage)
  })
})
