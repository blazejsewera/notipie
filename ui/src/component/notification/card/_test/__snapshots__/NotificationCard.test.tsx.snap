// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCard component tests renders correctly with full notification 1`] = `
<div
  className="rounded-3xl bg-white w-80 p-5 shadow-md relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <svg
        viewBox="0 0 1 1"
      >
        <rect
          fill="darkgreen"
          height={1}
          width={1}
        />
        <text
          dy="0.178"
          fill="#FFF"
          style={
            Object {
              "fontFamily": "Inter",
              "fontSize": 0.5,
              "fontWeight": "bold",
            }
          }
          textAnchor="middle"
          x={0.5}
          y={0.5}
        >
          Gi
        </text>
      </svg>
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800"
      >
        New Pull Request
      </p>
      <p
        className="text-xs font-bold text-gray-500"
      >
        — notipie
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-1.5 h-1.5 bg-yellow-500 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800"
  >
    <span>
      #12 add some new amazing functionality

Closes #10. Changes both in 'core' and 'ui'. Needs additional work with this and that.
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationCard component tests renders correctly with full with image notification 1`] = `
<div
  className="rounded-3xl bg-white w-80 p-5 shadow-md relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800"
      >
        New Pull Request
      </p>
      <p
        className="text-xs font-bold text-gray-500"
      >
        — notipie
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-1.5 h-1.5 bg-yellow-500 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800"
  >
    <span>
      #12 add some new amazing functionality

Closes #10. Changes both in 'core' and 'ui'. Needs additional work with this and that.
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationCard component tests renders correctly with minimal notification 1`] = `
<div
  className="rounded-3xl bg-white w-80 p-5 shadow-md relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <svg
        viewBox="0 0 1 1"
      >
        <rect
          fill="darkgreen"
          height={1}
          width={1}
        />
        <text
          dy="0.178"
          fill="#FFF"
          style={
            Object {
              "fontFamily": "Inter",
              "fontSize": 0.5,
              "fontWeight": "bold",
            }
          }
          textAnchor="middle"
          x={0.5}
          y={0.5}
        >
          Gi
        </text>
      </svg>
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800"
      >
        New Pull Request
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-1.5 h-1.5 bg-yellow-500 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800"
  >
    <span>
      
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationCard component tests renders correctly with partial notification 1`] = `
<div
  className="rounded-3xl bg-white w-80 p-5 shadow-md relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <svg
        viewBox="0 0 1 1"
      >
        <rect
          fill="darkgreen"
          height={1}
          width={1}
        />
        <text
          dy="0.178"
          fill="#FFF"
          style={
            Object {
              "fontFamily": "Inter",
              "fontSize": 0.5,
              "fontWeight": "bold",
            }
          }
          textAnchor="middle"
          x={0.5}
          y={0.5}
        >
          Gi
        </text>
      </svg>
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800"
      >
        New Pull Request
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-1.5 h-1.5 bg-yellow-500 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800"
  >
    <span>
      #12 add some new amazing functionality

Closes #10. Changes both in 'core' and 'ui'. Needs additional work with this and that.
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <div
        className="inline"
      >
        <svg
          className="stroke-current text-gray-500 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
